crates/ev-prover/src/prover/programs/batch.rs

```diff
@@ -18,23 +20,54 @@ use celestia_types::{
     nmt::{Namespace, NamespaceProof},
     Blob,
 };
+use ev_state_queries::hyperlane::indexer::HyperlaneIndexer;
+use ev_state_queries::DefaultProvider;
 use ev_types::v1::SignedData;
+use ev_zkevm_types::events::Dispatch;
 use ev_zkevm_types::programs::block::{BatchExecInput, BlockExecInput, BlockRangeExecOutput};
 use prost::Message;
 use reth_chainspec::ChainSpec;
 use rsp_client_executor::io::EthClientExecutorInput;
 use rsp_primitives::genesis::Genesis;
 use sp1_sdk::{include_elf, SP1ProofMode, SP1ProofWithPublicValues, SP1ProvingKey, SP1Stdin, SP1VerifyingKey};
+use storage::hyperlane::message::HyperlaneMessageStore;
 use tokio::{sync::mpsc, time::interval};
 use tracing::{debug, error, info, warn};
 
 use crate::config::Config;
 use crate::prover::ProgramProver;
 use crate::prover::{prover_from_env, SP1Prover};
 
+// Required because we want to pass a reference to the abi to calculate_batch_size
+type MailboxContractType = MailboxContract::MailboxContractInstance<
+    &'static alloy_provider::fillers::FillProvider<
+        alloy_provider::fillers::JoinFill<
+            alloy_provider::Identity,
+            alloy_provider::fillers::JoinFill<
+                alloy_provider::fillers::GasFiller,
+                alloy_provider::fillers::JoinFill<
+                    alloy_provider::fillers::BlobGasFiller,
+                    alloy_provider::fillers::JoinFill<
+                        alloy_provider::fillers::NonceFiller,
+                        alloy_provider::fillers::ChainIdFiller,
+                    >,
+                >,
+            >,
+        >,
+        alloy_provider::RootProvider,
+    >,
+>;
+
 /// The ELF (executable and linkable format) file for the Succinct RISC-V zkVM.
 pub const BATCH_ELF: &[u8] = include_elf!("ev-batch-program");
 
+sol! {
+    #[sol(rpc)]
+    contract MailboxContract {
```

Comment:
I think we should add this to a separate file under `prover` at ../

We could call it `abi.rs` to keep it minimal. Alternatively we could put this in the `chain.rs` file I've added in #298.